### ISSUE TITLE
10/9 Epicenter Admin & Epicenter

### ISSUE THREAD

**Issue by lzim** [lzim/teampsd#756]
Test whether putting mapped team members in “inactive” status removes them from the list of team members (and the estimate of team data) in the Facilitator Dashboard.

@staceypark 
- Inactivate Team 2 during the team session on Wednesday

@lzim 
- Inactivate Team 4 during the team session on Wednesday.

-----------------------------------------------------------------------------------------
**Workgroups leads**
**Identify Constraints (check all that apply):**
- [ ]   Capacity
- [ ]   Cost
- [ ]   Schedule
- [ ]   Interference
- [ ]   Capability
- [ ]   Other (describe)

_Add estimated person-hours to complete:_

**Edit the due date (if necessary)**

**Comment by staceypark** [comment]
I checked the facilitator dashboard, and unfortunately the test failed.
In that, if you make yourself "inactive" it is true that you aren't counted in the "denominator" of the attendance percentage. However, it looks like that denominator changes not by session but across all sessions.

i.e. In a 20 person team, if I make someone inactive, then the "denominator" for all of the sessions for that team goes down to 19.

FYI: @jamesmrollins 
Is this how the attendance is supposed to work  / is there any way to modify it?

**Comment by jamesmrollins** [comment]
@hirenp-waferwire please take a look below.  The team "Size" field appear to change when users are made inactive. The "Size" field value should be durable for that saved experiment.

**The original team size of "5"**

![image](https://user-images.githubusercontent.com/30132017/67172525-8254e800-f370-11e9-8bf5-8eb088312327.png)

**Then when two users made inactive in the Group directory, all change to "3".**

![image](https://user-images.githubusercontent.com/30132017/67172500-72d59f00-f370-11e9-9f34-d9bfc837b406.png)

**Comment by staceypark** [comment]
@jamesmrollins says Dev is already working on this and believes they will be able to get it done by Friday. @lzim As such, I think we should turn to training the IC's on Epicenter on Friday and start with Splashpage/data UI related functions first on Wednesday.

**Comment by lzim** [comment]
> @jamesmrollins says Dev is already working on this and believes they will be able to get it done by Friday. @lzim As such, I think we should turn to training the IC's on Epicenter on Friday and start with Splashpage/data UI related functions first on Wednesday.

Great! thanks!

**Comment by jamesmrollins** [comment]
@saveth and I discussed this issue below.  Saveth indicated that there are no report dependencies on this report.  We are going to change the Team Users field to only list active users.  We are going to change the label to say "Active Team Users:"

![image](https://user-images.githubusercontent.com/30132017/67448319-7d3fa500-f5cb-11e9-9e33-5ccea430c17d.png)

**Comment by staceypark** [comment]
@jamesmrollins Can you help me understand what I'm seeing here? How is it possible to have "200% attendance"?

![image](https://user-images.githubusercontent.com/31089501/71537026-12606200-28cb-11ea-833a-561176ae7a9f.png)

**Comment by jamesmrollins** [comment]
@staceypark - No, probably not.  I need to know the rest of the dynamics - what team-world were you logged into?

**Comment by staceypark** [comment]
@jamesmrollins 
100a1_abc_team_team

**Comment by jamesmrollins** [comment]
@hirenp-waferwire please take a look at the above calculations.  Should not exceed 100%.  Thanks, James

**Comment by jamesmrollins** [comment]
@staceypark the calculation changed because the number of active users recorded initially for that game was reduced, so it threw the percentage off.  Need a rule to keep active users active until a game is concluded, if you want to maintain a good percentage measurement.

**Comment by staceypark** [comment]
@jamesmrollins 
I see. Can you define "game conclusion" to me? Does a new "game" start after we hit "save" on an experiment"?
If not, will this number always be funky if someone part of that group, changes their role to facilitator to check the dashboard?

**Comment by jamesmrollins** [comment]
@staceypark "Game Conclusion" means that everyone stops adding to the instance.  Say, for instance, a sim is started, they fill out the hypothesis field, and run a base case and hit save.  The sim will take attendance and issue an attendance percentage calculated by the number of logins divided by the number of active users.  If only half the users login on subsequent session (without reset), and update another hypothesis field and hit save, then the attendance will be updated again, this time 50%.  So the facilitator will have to keep this in mind when orchestrating the sessions.  If this is not satisfactory, we may need to develop a new attendance feature to record it session by session (if that is possible).

**Comment by staceypark** [comment]
@jamesmrollins 
Hmmm..Can you check my understanding below?:

Say in a team of 10 = 8 learners and 2 facilitators
1. All 10 login. Fill out question and hypothesis filed and hit save.
2. Facilitator changes role in Epicenter to "facilitator" session. Because they already hit "save" the last piece of data should say "100%".
3. Next team meeting, some of the team is out.  6 people login, including both facilitators. They fill out findings and decisions from last meeting and hit save.
4. Facilitator changes role in Epicenter to "facilitator" after session. It should say "60%" now right? (As long as facilitator remembers after Step 2, to change back their role to "learner)?

You also said something about resetting above. I believe between experiments, learners would use the "save & reset". Does that mean the below would happen?
1. Only 6 of 10 login. This is a session that goes finishes experiment 1 and does the save & reset function to work on session 2. 
2. Since there were only 6 people when the reset happen it sounds like the "pie" of attendance would make 100% equal to 6 now.
3. In the next session all 10 people show up. They edit experiment 2 QHFD and hit save.
4. If the facilitator checks the attendance for this session in Step 3, would there be an attendance of over 100%?

**Comment by jamesmrollins** [comment]
@staceypark see below:

Algorithm:

1. Users login
2. Users interact with sim, click run, create experiment name and click "save"
3. System looks at number of logins divided by number of active users assigned to world001.
4. System records attendance against saved experiment name.
5. Users logoff
6. Users login
7. Users interact with sim and click "save"
8. System looks at number of logins divided by number of active users assigned to world001.
9. System updates attendance against saved experiment name
10. Users interact with sim and click "save and reset"
11. System updates attendance against saved experiment name
12. System closes experiment record. No other updates can happen.

Questions:

Say in a team of 10 = 8 learners and 2 facilitators. 
**I assume they are all logged in to a single world for a total of 10 either team leads or players. Whether a facilitator or a learner, the system sees either team leads or players at the world level.**

All 10 login. Fill out question and hypothesis filed and hit save.
**The sim will record 10 logins/10 world users for 100%**

Facilitator changes role in Epicenter to "facilitator" session. Because they already hit "save" the last piece of data should say "100%".
**Assuming you mean the facilitator unassigns themselves from the world. Correct, The sim won't take attendance until the next "save" event.**

Next team meeting, some of the team is out. 6 people login, including both facilitators. They fill out findings and decisions from last meeting and hit save.
**The sim will update record to 60%, provided facilitators are logged in as users assigned to the team's world**

Facilitator changes role in Epicenter to "facilitator" after session. It should say "60%" now right? (As long as facilitator remembers after Step 2, to change back their role to "learner)?
**If you mean they assign themselves back into a world as either a team lead or player, then yes.**

You also said something about resetting above. I believe between experiments, learners would use the "save & reset". Does that mean the below would happen?
**If the team hits "Save and Reset" then no more updates to the experiment are allowed.  The record is closed with current attendance.**

Only 6 of 10 login. This is a session that goes finishes experiment 1 and does the save & reset function to work on session 2.
Since there were only 6 people when the reset happen it sounds like the "pie" of attendance would make 100% equal to 6 now.
**The attendance would be 60%.  Because the denominator is set by the total number of people assigned to the world.  The numerator is the total number of people logged in.**

In the next session all 10 people show up. They edit experiment 2 QHFD and hit save.
**Then attendance would be 100%.**

If the facilitator checks the attendance for this session in Step 3, would there be an attendance of over 100%?
**The over 100% condition occurs if the world assignment list changes to be smaller than the number of active logins during the save event. **

**Comment by lzim** [comment]
@staceypark 

Is this being tracked int eh **document_tracker**?
I’ve added it to my list of things that should go in the expanded facilitator manual.

**Comment by staceypark** [comment]
@lzim I've cross-referenced this in the meta_document card, where we've been keeping track of all the team documentation we need to create.

If you have it tracked and I do, then I think we can close this issue?

**Comment by lzim** [comment]
> @lzim I've cross-referenced this in the meta_document card, where we've been keeping track of all the team documentation we need to create.
> 
> If you have it tracked and I do, then can I confirm we can close this issue?

Thanks for checking @staceypark 

This needs to be addressed in both the facilitator guides (detail) and the team how to (document).

**Comment by staceypark** [comment]
We can close this issue as 1) the original problem is no longer relevant with Facilitator Dashboard 2.0 and 2) documentation needs for Facilitator Dashboard 2.0 are tracked elsewhere.
